packages/apps/tasks/src/TaskList.tsx

```diff
@@ -0,0 +1,128 @@
+import { ShellLayout } from "@dxos/react-client";
+import { useIdentity } from "@dxos/react-client/halo";
+import { useQuery, useSpaces } from "@dxos/react-client/echo";
+import React, { useState } from "react";
+
+import { Task } from "./proto";
+import { useShell } from "@dxos/react-shell";
+
+export const TaskList = () => {
+  useIdentity({ login: true });
+  const [space] = useSpaces(); // What should the pattern be for find-or-create a space?
+  const shell = useShell();
+  const tasks = useQuery<Task>(space, Task.filter());
+  const [newTaskTitle, setNewTaskTitle] = useState("");
+  const [editingTask, setEditingTask] = useState<number | null>(null);
+  const [showDeleteTask, setShowDeleteTask] = useState<number | null>(null);
+
+  const handleNewTask = () => {
+    const task = new Task({ title: newTaskTitle, completed: false });
+    space.db.add(task);
+    setNewTaskTitle("");
+  };
+
+  return (
+    <div className="p-2">
+      <button
+        className="float-right bg-white hover:bg-gray-100 text-gray-800 font-semibold py-2 px-4 border border-gray-400 rounded shadow"
+        onClick={() =>
+          shell.setLayout(
+            ShellLayout.SPACE_INVITATIONS,
+            space?.key && { spaceKey: space.key }
+          )
+        }
```

Comment:
Is this canonical for how we want to access invitations?

Reply:
Currently, I think so yeah